### PR TITLE
[ShadowLayer] Support layer copying

### LINKED ISSUE #248: ShadowLayer does not implement initWithLayer:

### PR DESCRIPTION
**Pull request by romoore**
Implementing -initWithLayer: for Shadow layers

Closes #248

### CODE REVIEW

File: components/ShadowLayer/src/MDCShadowLayer.m [hunk 1]
```diff
@@ -145,6 +148,26 @@ - (void)commonMDCShadowLayerInit {
...
+  _topShadow.shadowColor = [UIColor blackColor].CGColor;
+
+  // -addSublayer: cannot be called from -initWithLayer: since it only works with model layers
+  [self addSublayer:_bottomShadow];
+  [self addSublayer:_topShadow];
+
+  [self commonMDCShadowLayerConfigure];
+}
+
 // TODO(#993): Implement missing initWithLayer:
```
Comment (ajsecord) [inline components/ShadowLayer/src/MDCShadowLayer.m]: Remove comment?

**Review by ajsecord — CHANGES_REQUESTED** [review]

File: components/ShadowLayer/src/MDCShadowLayer.m (hunk 1 above)
Comment (romoore) [reply components/ShadowLayer/src/MDCShadowLayer.m]: Done.

**Review by ajsecord — APPROVED** [review]

**Review by ianegordon — CHANGES_REQUESTED** [review]
commonMDCXYZInit is typically intended to be called from all init functions.  Adding commonMDCXYZConfig is new / different.  Do we really need to add this additional method?  Can we modify the other init methods so we only ever call commonMDCXYZInit?

**Review by ajsecord — APPROVED** [review]

**Comment by ajsecord** [comment]
@ianegordon your concerns were handled by Rob's latest commit.

**Review by ianegordon — APPROVED** [review]
Waiting for Travis, but LGTM.

**Comment by ianegordon** [comment]
Successfully tested locally on 8.x, 9.x, 10.x Simulators.

**Pull request merged**